desktop/core/src/desktop/js/apps/storageBrowser/FileChooserModal/FileChooserModal.tsx

```diff
@@ -0,0 +1,191 @@
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+import React, { useCallback, useEffect, useMemo, useState } from 'react';
+import Modal from 'cuix/dist/components/Modal';
+import { Input, Spin, Tooltip } from 'antd';
+import Table, { ColumnProps } from 'cuix/dist/components/Table';
+import classNames from 'classnames';
+
+import FolderIcon from '@cloudera/cuix-core/icons/react/ProjectIcon';
+import { FileOutlined } from '@ant-design/icons';
+
+import { i18nReact } from '../../../utils/i18nReact';
+import useDebounce from '../../../utils/useDebounce';
+import useLoadData from '../../../utils/hooks/useLoadData';
+
+import { PathAndFileData } from '../../../reactComponents/FileChooser/types';
+import { VIEWFILES_API_URl } from '../../../reactComponents/FileChooser/api';
+import PathBrowser from '../../../reactComponents/FileChooser/PathBrowser/PathBrowser';
+
+import './FileChooserModal.scss';
+
+interface FileChooserModalProps {
+  onClose: () => void;
+  onSubmit: (destination_path: string) => void;
+  showModal: boolean;
+  title: string;
+  sourcePath: string;
+  submitText?: string;
+  cancelText?: string;
+}
+
+interface FileChooserTableData {
+  name: string;
+  path: string;
+  type: string;
+}
+
+const FileChooserModal = ({
+  showModal,
+  onClose,
+  onSubmit,
+  title,
+  sourcePath,
+  ...i18n
+}: FileChooserModalProps): JSX.Element => {
+  const { t } = i18nReact.useTranslation();
+  const { cancelText = t('Cancel'), submitText = t('Submit') } = i18n;
+  const [destPath, setDestPath] = useState<string>(sourcePath);
+  const [searchTerm, setSearchTerm] = useState<string>('');
+
+  useEffect(() => {
+    setDestPath(sourcePath);
+  }, [sourcePath]);
+
+  const { data: filesData, loading } = useLoadData<PathAndFileData>(
+    `${VIEWFILES_API_URl}${destPath}`,
+    {
+      params: {
+        pagesize: '1000',
+        filter: searchTerm
+      },
+      skip: destPath === '' || destPath === undefined
+    }
+  );
+
+  const tableData: FileChooserTableData[] = useMemo(() => {
+    if (!filesData?.files) {
+      return [];
+    }
+
+    return filesData?.files?.map(file => ({
+      name: file.name,
+      path: file.path,
+      type: file.type
+    }));
+  }, [filesData]);
+
+  const handleSearch = useCallback(
+    useDebounce(searchTerm => {
+      setSearchTerm(encodeURIComponent(searchTerm));
+    }),
+    [setSearchTerm]
+  );
+
+  const getColumns = (file: FileChooserTableData) => {
+    const columns: ColumnProps<FileChooserTableData>[] = [];
```

Comment:
Are we not duplicating a lot this behaviour from the `StorageBrowserTable.tsx`?

In my opinion the `StorageBrowserTable.tsx` should be more modular, e.g. the actual table in it should be reusable component. I would see if it was possible to create a reusable and configurable "FilesystemTable" that could be used by different components (like this one) instead of duplicating this code. We will also need an actual "file picker" (for the importer and other places) which also needs a "FilesystemTable" to display the files and folders.